waiter/src/waiter/token_watch.clj

```diff
@@ -36,7 +39,7 @@
 (defn start-token-watch-maintainer
   "Starts daemon thread that maintains token watches and process/filters internal token events to be streamed to
   clients through the watch handlers. Returns map of various channels and state functions to control the daemon."
-  [kv-store clock tokens-update-chan-buffer-size channels-update-chan-buffer-size watch-refresh-timer-chan]
+  [kv-store clock tokens-update-chan-buffer-size channels-update-chan-buffer-size watch-refresh-timer-chan cid-factory-fn]
```

Comment:
Instead of passing a `cid-factory-fn` we can construct the CID internally in the code here using an iteration index or a timestamp. Not sure how passing the CID factory is helping.

Reply:
I was doing this originally to make it easier to assert events in unit tests. The ids are random and I'm not sure if there is a way to get the id value that utils/unique-identifier will return to make the test assertions

Reply:
You can mock functions with `with-redefs` for unit tests.

Reply:
start-token-watch-maintainer calls the `utils/unique-identifier` in an `async/go` block, which the `with-redef` can't mock functions. I can set a variable to equal the function outside the async go block and call that var which will be changed by the `with-redef`

Reply:
Is asserting on the exact value of the id important, or just asserting its presence sufficient?

Reply:
asserting the presence should be sufficient now that we are solely using unique ids for the event id instead of a request cid provided to us